### PR TITLE
Task-48160 : Can't close date picker when click outside or change page of send rewards page.

### PR DESCRIPTION
**Pull request by ahmadmido**
Issue: The problem is when clicking outside after opening the date picker or changing the page on the rewards page, the date picker component is still displayed and does not disappear. 
Solution: I have added an id for each component and a mounting method witch will control the date picker component for each click outside or changing page.

### CODE REVIEW

File: wallet-webapps-reward/src/main/webapp/vue-app/components/reward/SendRewardsTab.vue [hunk 1]
```diff
@@ -305,6 +307,8 @@ export default {
   data() {
     return {
       search: '',
+      id: `DatePicker${parseInt(Math.random() * 10000)}`,
```
Comment (rdenarie) [inline wallet-webapps-reward/src/main/webapp/vue-app/components/reward/SendRewardsTab.vue]: could you use a timestampt instead a random here ?
In addition should id and menuId have the same timestamp or not ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: wallet-webapps-reward/src/main/webapp/vue-app/components/reward/SendRewardsTab.vue (hunk 1 above)
Comment (ahmadmido) [reply wallet-webapps-reward/src/main/webapp/vue-app/components/reward/SendRewardsTab.vue]: > could you use a timestampt instead a random here ?
> In addition should id and menuId have the same timestamp or not ?

Done! It's not a problem I think when they have the same timestamp, the prefix is different ( DatePicker and DatePickerMenu)

**Review by rdenarie — CHANGES_REQUESTED** [review]
Can you add your PR description in the first commit message ? 
So that, it will be automatically used as commit message when you will make the squash

File: wallet-webapps-reward/src/main/webapp/vue-app/components/reward/SendRewardsTab.vue [hunk 2]
```diff
@@ -305,6 +307,8 @@ export default {
   data() {
     return {
       search: '',
+      id: `DatePicker${parseInt(new Date().toLocaleDateString())}`,
```
Comment (rdenarie) [inline wallet-webapps-reward/src/main/webapp/vue-app/components/reward/SendRewardsTab.vue]: This generate a date on format :

> console.log(event.toLocaleDateString(undefined, options));
// expected output: Thursday, December 20, 2012 (varies according to default locale)
I prefer that you use a timestamp with Date.now()

```suggestion
      id: `DatePicker${Date.now()}`,
```

File: wallet-webapps-reward/src/main/webapp/vue-app/components/reward/SendRewardsTab.vue [hunk 3]
```diff
@@ -305,6 +307,8 @@ export default {
...
+      menuId: `DatePickerMenu${parseInt(new Date().toLocaleDateString())}`,
```
Comment (rdenarie) [inline wallet-webapps-reward/src/main/webapp/vue-app/components/reward/SendRewardsTab.vue]: same here

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — CHANGES_REQUESTED** [review]
Code is ok.
Can you update the commit message : format should be 

<Titre>
<Ligne vide>
<Paragraphe pour la description> 

In your commit message, the title disappear, and you directly put the description

**Comment by ahmadmido** [comment]
> Code is ok.
> Can you update the commit message : format should be
> 
> In your commit message, the title disappear, and you directly put the description

Done!

**Review by rdenarie — CHANGES_REQUESTED** [review]
Can you add a new empty line between title and description in the commit message ?
You should have something like
<Title>
<Empty Line>
<Description>

**Review by rdenarie — APPROVED** [review]

**Pull request merged**